src/memray/commands/attach.py

```diff
@@ -298,6 +299,10 @@ def run(self, args: argparse.Namespace, parser: argparse.ArgumentParser) -> None
         verbose = args.verbose
 
         if args.method == "auto":
+            if platform.system() == "Linux" and debugger_available(
+                "gdb", verbose=verbose
+            ):
+                args.method = "gdb"
             if debugger_available("lldb", verbose=verbose):
```

Comment:
This needs to be `elif`, or we haven't actually changed the behavior at all - we'll still wind up preferring lldb on Linux.